types/persist/persist.go

```diff
@@ -70,7 +77,8 @@ func (p *Persist) Equals(p2 *Persist) bool {
 		p.LoginName == p2.LoginName &&
 		p.UserProfile == p2.UserProfile &&
 		p.NetworkLockKey.Equal(p2.NetworkLockKey) &&
-		p.NodeID == p2.NodeID
+		p.NodeID == p2.NodeID &&
+		reflect.DeepEqual(p.DisallowedTKAStateIDs, p2.DisallowedTKAStateIDs)
```

Comment:
do you care about nil vs empty slice? this treats them differently, but I doubt that matters. you might want to use that len-zero-map-to-nil func we have somewhere

Reply:
```
bradfitz@tsdev:~/src/tailscale.com$ git grep ^func | grep nil | grep any
wgengine/router/ifconfig_windows_test.go:func nilIfEmpty[E any](s []E) []E {
```

Reply:
Done.